### PR TITLE
Forward merge 2.1 into develop, with some changes

### PR DESCRIPTION
**Pull request by JeetKunDoug**
Forward merges all 2.0/2.1 changes to develop with a few changes:
- 2.1 reverted 'remove legacy gossip' as it made some breaking API changes and removed functionality required by riak_repl. This merge maintains just the riak_repl required changes, while allowing the rest of the legacy gossip changes to remain.
- updates rebar.config to use develop branches of riak_sysmon and riak_ensemble

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -13,10 +13,10 @@
   {lager, "2.0.3", {git, "git://github.com/basho/lager.git", {tag, "2.0.3"}}},
   {poolboy, ".*", {git, "git://github.com/basho/poolboy.git", {tag, "0.8.1p3"}}},
   {basho_stats, ".*", {git, "git://github.com/basho/basho_stats.git", {tag, "1.0.3"}}},
-  {riak_sysmon, ".*", {git, "git://github.com/basho/riak_sysmon.git", {tag, "2.0.1"}}},
-  {riak_ensemble, ".*", {git, "git://github.com/basho/riak_ensemble", {tag, "2.1.0"}}},
+  {riak_sysmon, ".*", {git, "git://github.com/basho/riak_sysmon.git", {branch, "develop"}}},
```
Comment (nickelization) [inline rebar.config:16]: You said something earlier about possibly splitting these branch changes off into a separate PR or commit. Still want to do that or should we just continue as is?

Comment (JeetKunDoug) [inline rebar.config:16]: Yea - figured fixing the rebar.config as part of the forward merge, rather than breaking develop, was reasonable.

**Comment by nickelization** [comment]
:+1: c630abc

**Comment by JeetKunDoug** [comment]
@borshop merge

**Pull request merged**